### PR TITLE
feat: adding script for creating local mas-sso instance

### PR DESCRIPTION
**Pull request by akoserwal**
Adding a script to create a local mas-sso using keycloak container with required realm config.

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has created for changes required on the client side

### CODE REVIEW

File: scripts/mas_sso.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+set -e
+
+docker run -d -p 8180:8080 -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin \
```
Comment (machi1990) [inline scripts/mas_sso.sh:5]: The user is now `admin` and the password is  `admin`?

Comment (machi1990) [inline scripts/mas_sso.sh:5]: let's also configure this as GH actions services: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/11c964a74facabc3d7c1a121e3984ffd3eb3c53b/.github/workflows/ci.yaml#L70

**Review by machi1990 — COMMENTED** [review]
Left some comments on the script part.

A general question, midstream pipelines should be fine and they can continue to run against mas-sso. Is this something we want the pipelines to stop using? They are current configured under these docker files:https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/tree/11c964a74facabc3d7c1a121e3984ffd3eb3c53b/docker

File: scripts/mas_sso.sh (hunk 1 above)
Comment (akoserwal) [reply scripts/mas_sso.sh:5]: Just for testing, we can even skip setting up the username/password.

**Comment by akoserwal** [comment]
> https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/tree/11c964a74facabc3d7c1a121e3984ffd3eb3c53b/docker

yes, I think we should stop using it even in the midstream because the current test leaks lot of clients in mas-sso. Which aren't getting cleaned up properly. it will create an issue for the migration and I don't think sso will support it.

File: internal/kafka/test/integration/serviceaccounts_test.go [hunk 2]
```diff
@@ -269,57 +269,57 @@ func TestServiceAccounts_InputValidation(t *testing.T) {
...
-	Expect(resp.StatusCode).To(Equal(http.StatusForbidden))
-
-	//cleanup
-	_, _, err = client.SecurityApi.DeleteServiceAccountById(ctx, sa.Id)
-	Expect(err).ShouldNot(HaveOccurred())
-
-	_, _, err = client.SecurityApi.DeleteServiceAccountById(ctx, sa2.Id)
-	Expect(err).ShouldNot(HaveOccurred())
-}
+//Todo Temporary commenting out the test
```
Comment (pb82) [inline internal/kafka/test/integration/serviceaccounts_test.go:272]: can we enable this test?

**Comment by pawelpaszki** [comment]
integration tests still fail: 

```
failed to get new token from gocloak with error: could not get token: Post "http://127.0.0.1:8180/auth/realms/rhoas/protocol/openid-connect/token": dial tcp 127.0.0.1:8180: connect: connection refused: KAFKAS-MGMT-9: failed to create service account for cluster 2aad9fc1-c40e-471f-8d57-fdaecc7d3686 due to error: KAFKAS-MGMT-9: failed to register agent service account
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -92,8 +102,12 @@ jobs:
         uses: actions/setup-go@v2
         with:
           go-version: "1.16"
+      - name: keycloak service volume permission setup
+        run:  sudo chown -R 1001:1001 $GITHUB_WORKSPACE
```
Comment (machi1990) [inline .github/workflows/ci.yaml:106]: This might not be needed anymore

File: scripts/mas-sso-config.sh [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+KAS=`curl -sk --header "Content-Type: application/json" --header "Authorization: Bearer $TOKEN" $KEYCLOAK_URL/auth/admin/realms/rhoas-kafka-sre/clients?clientId=kas-fleet-manager`
+kasClientId=$(jq -r '.[].id' <<< $KAS)
+
+#/auth/admin/realms/rhoas/clients/de121cf7-a6b2-4d39-a99c-8da787454a66/service-account-user
+SVC=`curl -sk --header "Content-Type: application/json" --header "Authorization: Bearer $TOKEN" $KEYCLOAK_URL/auth/admin/realms/rhoas-kafka-sre/clients/$kasClientId/service-account-user`
+svcUserId=$(jq -r '.id' <<< $SVC)
+echo $svcUserId
+
+FINAL=`curl -sk --data-raw '[{"id": '$manageUser',"name": "manage-users"},{"id": '$manageRealm',"name": "manage-realm"},{"id": '$manageClients',"name": "manage-clients"}]' --header "Content-Type: application/json" --header "Authorization: Bearer $TOKEN" $KEYCLOAK_URL/auth/admin/realms/rhoas-kafka-sre/users/$svcUserId/role-mappings/clients/$realmMgmtClientId`
```
Comment (ziccardi) [inline scripts/mas-sso-config.sh:111]: $( .. ) syntax should be preferred to back tick syntax (https://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_xcu_chap02.html#tag_23_02_06_03)

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (akoserwal) [reply .github/workflows/ci.yaml]: removed

File: scripts/mas-sso-config.sh (hunk 4 above)
Comment (akoserwal) [reply scripts/mas-sso-config.sh]: Thanks for the suggestion. we can improve this later. Anyways this part will be changed with migration.

**Comment by akoserwal** [comment]
@pb82 @pawelpaszki @machi1990: Integration test passed

**Comment by machi1990** [comment]
> @pb82 @pawelpaszki @machi1990: Integration test passed

lgtm, thanks a lot for the great work.

Let's sync up to make sure that the integration test in midstream won't be affected by the change.

File: scripts/mas_sso_setup.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+set -e
+
+docker network create mas-sso-network || true
+
+docker run \
+  --name=mas-sso \
+  --net mas-sso-network \
+  -p 8180:8080 \
+  -e KEYCLOAK_USER=admin \
```
Comment (machi1990) [inline scripts/mas_sso_setup.sh:11]:
```suggestion
  -e DB_VENDOR=h2  \
  -e KEYCLOAK_USER=admin \
```

File: scripts/mas_sso_teardown.sh [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+# Additionally, because the password for the user is passed through the command line, its possible
+# for other users on the system to see the password in `ps` output.
+
+set -e
+
+docker stop mas-sso
+
+docker rm mas-sso
+
+docker network rm mas-sso-network
```
Comment (machi1990) [inline scripts/mas_sso_teardown.sh:15]:
```suggestion
docker network rm mas-sso-network

```

**Review by machi1990 — APPROVED** [review]
lgtm

The remaining CI failure will be fixed by https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/839

**Pull request merged**